custom/enikshay/tasks.py

```diff
@@ -156,17 +164,24 @@ def run_batch(self, case_ids):
         error_count = 0
         success_count = 0
         case_batches = 0
+        ledger_batches = 0
 
         errors = []
         with Timer() as t:
             batch_size = 100
-            updates = []
+            case_updates = []
+            ledger_updates = []
             for episode in self._get_open_episode_cases(case_ids):
                 did_error = False
                 update_json = {}
                 for updater in self.updaters:
                     try:
-                        potential_update = updater(self.domain, episode).update_json()
+                        _updater = updater(self.domain, episode)
+                        potential_update = _updater.update_json()
+                        if isinstance(_updater, EpisodeAdherenceUpdate):
+                            episode_ledger_updates = _updater.ledger_updates()
+                            if episode_ledger_updates:
```

Comment:
I think you can just call `extend` here directly without checking if there are updates since:
```py
In [1]: x = ['a']

In [2]: x.extend([])

In [3]: x
Out[3]: ['a']
```